### DISCUSSION TITLE
index columns / show rows

### DISCUSSION THREAD

**Discussion by mpapis** [activeadmin/activeadmin#8135] [Ideas]
I'm using:
```ruby
index do
  columns :id, ...
end
show do
  rows :id, ...
  table_for ... do
    columns ...
  end
end
```
I know `rows` and `columns` is **NOT** documented, but it's pretty useful, especially with decorators where the show.rows and index.columns only define what to show and decorators define how to show it (I'm using mine [ArbreDecorator](https://github.com/mpapis/activeadmin_decorator/tree/master#arbredecorator)). 

To make `columns` work - I had to do:
```ruby
Arbre::Element::BuilderMethods.remove_method :columns
```

Now the question arises, is it worth documenting show.rows/index.columns and the trick to make columns work.

Going a step further could you consider extracting `ActiveAdmin::Views::Columns` to separate gem so it does not have to disabled but rather included with a note it affects index.columns behavior?

Optionally the `Arbre::Element::BuilderMethods.columns` method could detect if it's first (and second?) argument is a symbol and act as `ActiveAdmin::Views::TableFor.columns` otherwise act as `ActiveAdmin::Views::Columns`.

I can help out with any changes needed :)

WDYT?

**Comment by javierjulio** [comment]
If I understand right, yes, I can see how this is an issue. Thankfully though it's been resolved by removing the Columns component entirely for v4 which is coming soon. You shouldn't have an issue anymore with that release going forward. Thank you.

I've removed many builder methods so you should have less conflicts. For example, we declared our site header and footer components as `header` and `footer` but that conflicts with the HTML elements of the same name, so if you tried to use it (through Arbre) it would attempt to render ActiveAdmin's site header or footer components. These issues have been resolved with v4 and noted in the upgrading guide.
